### PR TITLE
Update Agent to 1.29.1

### PR DESCRIPTION
**Pull request by shubham2892**
### Summary

Update Agent to 1.29.1

### Testing

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: allow-trunk-api-policy [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (suneyz) [inline allow-trunk-api-policy]: Is this supposed to be in the update version commit?

File: scripts/test-publish-docker-images [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (suneyz) [inline scripts/test-publish-docker-images]: Is this supposed to be in the update version commit?

File: allow-trunk-api-policy (hunk 1 above)
Comment (shubham2892) [reply allow-trunk-api-policy]: No this should not be. Will fix.

File: scripts/test-publish-docker-images (hunk 2 above)
Comment (shubham2892) [reply scripts/test-publish-docker-images]: No this should not be.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,10 @@
 # Changelog
+## 1.29.1
```
Comment (fenxiong) [inline CHANGELOG.md]: can you add this (from https://github.com/aws/amazon-ecs-agent/pull/2093):
Bug: update amazon-vpc-cni-plugins to include a fix (https://github.com/aws/amazon-vpc-cni-plugins/commit/fa071b269ea57644ab566cc915698c61369e158f) that allows awsvpcTrunking to work without ec2-net-utils.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,10 @@
...
+* Feature - Update task cleanup wait logic to clean task resources immediately instead of waiting 3 hours [2084](https://github.com/aws/amazon-ecs-agent/pull/2084)
```
Comment (fenxiong) [inline CHANGELOG.md]: this should be enhancement i think. otherwise you will need major rather than minor version bump

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,11 @@
 # Changelog
+## 1.29.1
+* Enhancement - Update task cleanup wait logic to clean task resources immediately instead of waiting 3 hours [2084](https://github.com/aws/amazon-ecs-agent/pull/2084)
```
Comment (fenxiong) [inline CHANGELOG.md]: missing a "#" before 2084

Comment (shubham2892) [reply CHANGELOG.md]: Good catch.

**Pull request merged**